docs/README.md

```diff
@@ -0,0 +1,38 @@
+# Nomad Codebase Documentation
+
+This directory contains some documentation about the Nomad codebase,
+aimed at readers who are interested in making code contributions.
+
+If you're looking for information on _using_ using, please instead refer
+to [the main Nomad website](https://nomadproject.io).
+
+## Architecture
+
+The code for Nomad's major components is organized as:
+
+* `api/` provides a Go client for Nomad's HTTP API.
+* `client/` contains Nomad's client agent code.
+* `command/` contains Nomad's CLI code.
+* `nomad/` contains Nomad's server agent code.
+
+The high level control flow for many Nomad commands are:
+
+```
+# Read commands:
+CLI command -> HTTP API -> RPC -> StateStore
```

Comment:
Perhaps replace CLI with client? Unless the UI also relies on CLI command library and everything goes through that. I think it would be important to track both, since they both read & change state, but may have different mechanisms for doing so.